projects/drivers/axi_adc_core/axi_adc_core.h

```diff
@@ -52,12 +52,16 @@ struct axi_adc {
 	uint32_t base;
 	uint8_t	num_channels;
 	uint64_t clock_hz;
+	int32_t (*axi_io_read)(uint32_t, uint32_t, uint32_t*);
```

Comment:
why does ad9528 has the extra parameter and here we have the read/write function prototype ?

Reply:
The `extra` void pointer is used to point at platform specific parameters (i.e. Xilinx/Altera spi config params) while `axi_io_read` and `axi_io_write` are pointers to functions that wrap read/write platform specific functions for AXI IO interface.